pyanaconda/payload/dnfpayload.py

```diff
@@ -1291,8 +1291,12 @@ def _add_treeinfo_repositories(self, install_tree_url, base_repo_url=None):
 
             for repo_md in self._install_tree_metadata.get_metadata_repos():
                 if repo_md.path not in existing_urls:
-                    repo = RepoData(name=repo_md.name, baseurl=repo_md.path,
-                                    install=False, enabled=True)
+                    if conf.payload.enable_addon_repos:
+                        repo = RepoData(name=repo_md.name, baseurl=repo_md.path,
+                                        install=False, enabled=True)
+                    else:
+                        repo = RepoData(name=repo_md.name, baseurl=repo_md.path,
+                                        install=False, enabled=False)
```

Comment:
Please, create a new instance of `RepoData` with `enabled=conf.payload.enable_addon_repos` to avoid the branching.

Reply:
I'll make this change shortly.

Reply:
Again, please avoid the branching. For example, you can use a new variable:
`repo_enabled = repo_treeinfo.type in enabled_repositories_from_treeinfo`